html/modules/custom/reliefweb_guidelines/src/Controller/GuidelineSinglePageController.php

```diff
@@ -0,0 +1,429 @@
+   *   The reliefweb api client service.
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   *   The renderer service.
+   * @param \Drupal\Core\State\StateInterface $state
+   *   The state service.
+   */
+  public function __construct(
+    AccountProxyInterface $current_user,
+    EntityTypeManagerInterface $entity_type_manager,
+    ReliefWebApiClient $reliefweb_api_client,
+    RendererInterface $renderer,
+    StateInterface $state
+  ) {
+    $this->currentUser = $current_user;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->reliefWebApiClient = $reliefweb_api_client;
+    $this->renderer = $renderer;
+    $this->state = $state;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('current_user'),
+      $container->get('entity_type.manager'),
+      $container->get('reliefweb_api.client'),
+      $container->get('renderer'),
+      $container->get('state')
+    );
+  }
+
+  /**
+   * Get the page content.
+   *
+   * @return array
+   *   Render array for the homepage.
+   */
+  public function getPageContent() {
+    $storage = $this->entityTypeManager->getStorage('guideline');
+
+    $ids = $storage
+      ->getQuery()
+      ->sort('id', 'ASC')
+      ->execute();
+
+    /** @var \Drupal\guidelines\Entity\Guideline[] $guidelines */
+    $guidelines = $storage->loadMultiple($ids);
+
+    $items = [];
+
+    foreach ($guidelines as $guideline) {
+      if ($parents = $guideline->getParentIds()) {
+        $items[$parents[0]]['#children'][] = [
+          '#theme' => 'reliefweb_guidelines_item',
+          '#id' => $guideline->hasField('field_short_link') ? Html::getUniqueId($guideline->field_short_link->value) : Html::getUniqueId($guideline->field_title->value),
+          '#title' => $guideline->field_title->value,
+          '#description' => $guideline->hasField('field_description') ? check_markup($guideline->field_description->value, $guideline->field_description->format) : '',
+        ];
+      }
+      else {
+        $items[$guideline->id()] = [
+          '#theme' => 'reliefweb_guidelines_item',
+          '#id' => $guideline->hasField('field_short_link') ? Html::getUniqueId($guideline->field_short_link->value) : Html::getUniqueId($guideline->field_title->value),
+          '#title' => $guideline->field_title->value,
+          '#description' => $guideline->hasField('field_description') ? check_markup($guideline->field_description->value, $guideline->field_description->format) : '',
+          '#children' => [],
+        ];
+      }
+    }
+
+    $build = [
+      '#theme' => 'reliefweb_guidelines_list',
+      '#title' => $this->t('ReliefWeb guidelines'),
+      '#guidelines' => $items,
+      '#cache' => [
+        'contexts' => ['user.permissions'],
+      ],
+    ];
+
+    $build['#attached']['library'][] = 'reliefweb_guidelines/reliefweb-guidelines';
+
+    return $build;
+  }
+
+  /**
```

Comment:
Looks like all the methods below are leftovers from copying/pasting and can be removed.